### ISSUE TITLE
mtcp_readfile() -- we should check its return address

### ISSUE THREAD

**Issue by gc00** [dmtcp/dmtcp#47]
In mtcp/mtcp_restart.c, it calls mtcp_readfile().  Recently, I modified the definintion to follow the semantics of the read() system call (0 = end of file, -1 = error, etc.).  If it detects -1 and
mtcp_sys_errno is EAGAIN or EINTR, it does a retry, and after 10 retries, it aborts
For any other error, it aborts.  Hence, currently, the caller of mtcp_readfile() mostly doesn't check the return value.  We should consider checking the return value.  (But if it returns -1, note that mtcp_sys_errno was available only as a local variable inside mtcp_readfile() for certain technical reasons.  We don't want to use global variables in mtcp_restart, and so it's not clear how we would pass errno to the outside world.

**Comment by gc00** [comment]
This was fixed in commit 0d3b22b3841d2781025c68aa2da4450a71bbb22d .

**Comment by karya0** [comment]
There are still some places where we do not check the return value of `mtcp_readfile`

**Comment by gc00** [comment]
That's true.  But is it important to check the return value?  Currently, mtcp_readfile() can only return 0 (EOF) or the full 'size' that was requested to be read.  In any other case, it aborts instead of returning -1 (error).  Do we need to check for a return value of 0 or positive in the other locations in mtcp_restart.c?
